application/src/test/java/org/thingsboard/server/controller/AbstractWebTest.java

```diff
@@ -699,4 +711,49 @@ protected <T extends HasId<? extends UUIDBased>> ListenableFuture<List<ResultAct
         return Futures.allAsList(futures);
     }
 
+    protected void tesEntityDaoWithRelationsOk(EntityId entityIdFrom, EntityId entityTo, String urlDelete) throws Exception {
+        createEntityRelation(entityIdFrom, entityTo, "TEST_TYPE");
+        assertThat(getRelationsTo(entityTo)).hasSize(1);
+
+        doDelete(urlDelete).andExpect(status().isOk());
+
+        assertThat(getRelationsTo(entityTo)).hasSize(0);
+    }
+
+    protected <T> void tesEntityDaoWithRelationsTransactionalException(Dao<T> dao, EntityId entityIdFrom, EntityId entityTo,
+                                                                       String urlDelete) throws Exception {
+        entityDaoRemoveByIdWithException (dao);
+        createEntityRelation(entityIdFrom, entityTo, "TEST_TRANSACTIONAL_TYPE");
+        assertThat(getRelationsTo(entityTo)).hasSize(1);
+
+        doDelete(urlDelete)
+                .andExpect(status().isInternalServerError());
+
+        assertThat(getRelationsTo(entityTo)).hasSize(1);
+    }
+
+    protected <T> void entityDaoRemoveByIdWithException (Dao<T> dao) {
```

Comment:
remove from abstract. remove method.
leave just one line in the test code. This code is essential for understanding the test for the reader.

Reply:
No